src/recv_request_vote_result.c

```diff
@@ -50,6 +51,12 @@ int recvRequestVoteResult(struct raft *r,
         }
     }
 
+    /* Converted to follower as a result of seeing a higher term. */
+    if (r->state != RAFT_CANDIDATE) {
+        tracef("no longer candidate -> ignore");
+        return 0;
```

Comment:
This one is a bit hard to write a test for, I added it because we should not be accessing `r->candidate_state` anymore on line 87 when we're no longer candidate as a result of `recvEnsureMatchingTerms` on line 48.

Reply:
I don't recall the details, but maybe at that point we are sure we are candidate? If that's the case you could replace the `if` with an assertion.

Reply:
I think I see what you mean. I can see a test for that, but yes it requires some work. It's okay if you don't feel to spend time on that.